### DISCUSSION TITLE
Poor results on Pi 4

### DISCUSSION THREAD

**Discussion by virtual812** [owncast/owncast#1971] [Q&A] [answered]
Hi, just fired up Owncast on a Pi using the install script.
Used the suggested Jan 2021 image.
Got it all up and running and had questionable performance until i realised OMX was not selected by default.
Still not getting the performance we need.

Have a use case where i want to give on some occasions and see on other occasions a high res low latency stream.
Essentially watching a colleague working with CAD and guiding the design.

We're hoping for 2560x1440 / 30FPS / 3000 - 6000 Kbps with low latency.

The network is capable, no issue there, solid low latency up and downstream either end.

In our testing as soon as anything over about 1500Kbps or around 720-960p is streamed (NVENC from OBS, no scaling of res or FPS in Owncast) we start to see high amounts of frame drops, and latency is around the 10 sec mark with the slider at the second lowest latency position.

Single client viewing on Owncast during testing.

So the ultimate question...
Is this pushing the limits of the Pi4 or is there a chance that a misconfiguration could exist?

**Comment by m0ukd** [comment, marked as answer]
If it's just for a colleague to watch and you don't need all the chat functionality and goodness of Owncast, I would recommend rtsp-simple-server at https://github.com/aler9/rtsp-simple-server

You can publish via RTMP the same as you do with Owncast, but latency is around a second (no re-encoding). You can then watch with VLC as RTMP, RTSP, even UDP but most useful and simple, directly in a browser with HLS on port 8888 👍

**Reply by virtual812** [reply]
Thanks for the info, this might be just what we want.

Second question within a question, is the Pi just simply performance limited or is there a chance it could have done the job?

**Reply by gabek** [reply]
I've certainly seen Raspberry Pis work just great with Owncast, especially given it has hardware accelerated encoding. Even without the OMX codec I've seen people use a Pi with Owncast. Unfortunately I don't know enough about your setup to make any changes or recommendations,

**Reply by ghost** [reply]
"We're hoping for 2560x1440 / 30FPS / 3000 - 6000 Kbps with low latency."

You're expecting WAY WAY WAY too much from a $40 credit card sized PC.

**Reply by virtual812** [reply]
> You're expecting WAY WAY WAY too much from a $40 credit card sized PC.

If the workload fell to the CPU then yeah, it's too much to ask.
I however have heard good things about the GPU acceleration on the Pi and thought that might get us over the line.
